### PR TITLE
P/brentpicasso/lua bitop 2.9.x

### PR DESCRIPTION
**Pull request by brentpicasso**
* Enable bit operations for Lua
* Convert lua number to double, which has the side effect of fixing 29 bit CAN IDs.

### CODE REVIEW

**Comment by stieg** [comment]
Addresses #185 and #376

File: lib_lua/src/luaconf.h [hunk 1]
```diff
@@ -506,9 +506,12 @@
 #if defined(LUA_NUMBER_INTEGER)
 #define LUA_NUMBER      int
 #else
-#define LUA_NUMBER      float
+#define LUA_NUMBER      double
 #endif
 
+#define LUA_NUMBER double
+#define LUA_NUMBER_DOUBLE LUA_NUMBER
```
Comment (stieg) [inline lib_lua/src/luaconf.h]: Why is this duplicated here?

Comment (brentpicasso) [inline lib_lua/src/luaconf.h]: bitOp wants to see LUA_NUMBER_DOBULE defined for some reason.

File: lib_lua/src/luaconf.h [hunk 2]
```diff
@@ -527,7 +530,7 @@
 #if defined(LUA_NUMBER_INTEGER)
 #define lua_number2str(s,n)     modp_itoa10((n),s)
 #else
-#define lua_number2str(s,n)     modp_ftoa((n),s, LUA_NUMBER_PRECISION)
+#define lua_number2str(s,n)     modp_dtoa((n),s, LUA_NUMBER_PRECISION)
```
Comment (stieg) [inline lib_lua/src/luaconf.h]: We should use the stdlib functions as we are using those now and will be getting rid of the modp methods in a later release

Comment (stieg) [inline lib_lua/src/luaconf.h]: Err.. by stdlib I mean the methods in string.h that will convert a double to a char array.

Comment (brentpicasso) [inline lib_lua/src/luaconf.h]: for 2.9.x or the later release? unclear.

File: lib_lua/src/luaconf.h (hunk 1 above)
Comment (stieg) [inline lib_lua/src/luaconf.h]: I mean why the re-definition in line 512?

**Comment by stieg** [comment]
Minor nits.  Address and :shipit:

File: lib_lua/src/luaconf.h (hunk 1 above)
Comment (brentpicasso) [inline lib_lua/src/luaconf.h]: left over cruft. fixed.

File: lib_lua/src/luaconf.h (hunk 2 above)
Comment (brentpicasso) [inline lib_lua/src/luaconf.h]: I'll save the modp_xxxx for a holistic sweep through when we do it in a later release.

**Pull request merged**